infra/networks/subnet-backfill.tf

```
@@ -0,0 +1,92 @@
+# The purpose of this file is to backfill the default VPC with 3 private subnets.
+
+locals {
+  backfill_subnet_cidrs = {
+    # The CIDRs were chosen to be within `172.31.0.0/16` but not overlap with the nearest
+    # CIDRs already being used in the VPC.
+    #
+    # You can can confirm the ranges with a tool like:
+    # https://www.ipaddressguide.com/cidr
+    #
+    # The `/20` CIDR block was chosen because most of the existing subnets are `/20`.
+    "us-east-1a" = "172.31.144.0/20", # /20 = 4096 IPs, last address is 172.31.159.255
+    "us-east-1b" = "172.31.160.0/20", # /20 = 4096 IPs, last address is	172.31.175.255
+    "us-east-1c" = "172.31.176.0/20", # /20 = 4096 IPs, last address is 172.31.191.255
+  }
+}
+
+# ------- #
+# SUBNETS #
+# ------- #
+
+# docs: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/subnet
+resource "aws_subnet" "backfill_private" {
+  count                   = length(local.backfill_subnet_cidrs)
+  vpc_id                  = data.aws_vpc.default.id
+  availability_zone       = keys(local.backfill_subnet_cidrs)[count.index]
+  cidr_block              = values(local.backfill_subnet_cidrs)[count.index]
+  map_public_ip_on_launch = false
+  tags = {
+    Name        = "backfill-private-${count.index}"
+    subnet_type = "private"
+  }
+}
```

Would it be clearer to use `for_each`? Something like this (untested):

```suggestion
resource "aws_subnet" "backfill_private" {
  for_each                = local.backfill_subnet_cidrs
  vpc_id                  = data.aws_vpc.default.id
  availability_zone       = each.key
  cidr_block              = each.value
  map_public_ip_on_launch = false
  tags = {
    Name        = "backfill-private-${each.key}"
    subnet_type = "private"
  }
}
```

---

I tried that, and it didn't quite work. I'll try it again and see how it goes 🤞🏼